src/sysroot.rs

```diff
@@ -112,13 +118,16 @@ version = "0.0.0"
         }
 
         // Copy artifacts to Xargo sysroot
-        util::cp_r(
-            &td.join("target")
-                .join(cmode.triple())
-                .join(profile())
-                .join("deps"),
-            &dst,
-        )?;
+        match target_dir {
+            Some(dir) => util::cp_r(
+                &config.unwrap().root().join(dir).join(cmode.triple()).join(profile()).join("deps"),
+                &dst
+            ),
+            None      => util::cp_r(
+                &td.join("target").join(cmode.triple()).join(profile()).join("deps"),
+                &dst
+            ),
```

Comment:
There's a lot of shared code between the two branches here, could you factor that out?

Reply:
That's true, I'll improve that.